packages/SwingSet/docs/virtual-objects.md

```diff
@@ -15,118 +15,116 @@ A virtual object has a "kind", which defines what sort of behavior and state it
 
-- The in-memory manifestations of a virtual object (which we call its "representatives") are not necessarily (or even usually) `===` to each other, even when they represent the same object, since a new representative is created for each act of deserialization.  They do, however, all operate on the same state, so changes to the state made by a method call to one representative will be visible to the other representatives.  A soon-to-arrive future version of the virtual object system will add a `sameKey` predicate that will allow you to compare two virtual object representatives to determine if they refer to the same virtual object instance.
+- The in-memory manifestations of a virtual object (which we call its "representatives") are not necessarily (or even usually) `===` to each other, even when they represent the same object, since a new representative is created for each act of deserialization.  They do, however, all operate on the same state, so changes to the state made by a method call to one representative will be visible to the other representatives.
 
-- The object returned by the `bodyMaker` function _is_ the representative, so you can capture it inside any of the body methods (including `initialize`) in order to pass the virtual object to somebody else.  So you can write things like the following:
+- The `self` object returned by the `instanceKitMaker` function _is_ the representative, so you can capture it inside any of the body methods (as well as the `init` function) in order to pass the virtual object to somebody else.  So you can write things like the following:
 
 ```javascript
-  function thingBody(state) {
+  function thingKitMaker(state) {
     const self = {
-      initialize(someParam) {
-        otherObject.doSomethingWithMe(self, someParam);
-        state.foo = 47;
-      },
       sendMeTo(somebodyElse) {
         somebodyElse.hereIAm(self);
       },
       ... // and so on
     };
-    return self;
+    function initialize(someParam) {
+      otherObject.doSomethingWithMe(self, someParam);
+      state.foo = 47;
+    },
+    return { init, self };
   }
 ```
 
 ## Persistent Weak Stores
 
 The vat secondary storage system also includes a `makeWeakStore` function, which is identical in API to the `makeWeakStore` function provided by the `@agoric/store` package.  I.e., you call:
 
-`const ws = makeWeakStore();`
+`  const ws = makeWeakStore();`
 
 to obtain a new weak store instance, which you can then access using the `has`, `init`, `set`, `get`, and `delete` methods just as you would if you had imported `makeWeakStore` from `@agoric/store`.
 
 However, the vat secondary storage system's `makeWeakStore` augments the one provided by the `@agoric/store` package in two important ways:
```

Comment:
Another important caveat would reiterate that the values associated with representatives get hardened and serialized immediately.